app/src/main/java/com/bc/gordiansigner/model/HDKey.kt

```diff
@@ -0,0 +1,63 @@
+package com.bc.gordiansigner.model
+
+import com.bc.gordiansigner.helper.Network
+import com.blockstream.libwally.Wally
+import com.blockstream.libwally.Wally.*
+
+class HDKey {
+
+    private val rootKey: Any
+
+    constructor(key: Any, network: Network) {
+        rootKey = key
+        this.network = network
+    }
+
+    constructor(seed: ByteArray, network: Network) {
+        this.network = network
+        rootKey = bip32_key_from_seed(
```

Comment:
I prefer to move all kind of stuff like this to the lower-level class like `Bip32`. It will be more clear and easier to write the tests. Then you can call `Bip32.rootKeyFromSeed`, something like that.